--- old/tools/tools.pro	2014-12-05 17:25:10.000000000 +0100
+++ new/tools/tools.pro	2014-12-21 12:28:26.813610634 +0100
@@ -1,20 +1,11 @@
 TEMPLATE = subdirs
-SUBDIRS += \
-    qmlmin \
-    qmlimportscanner
 
 qmlmin.CONFIG = host_build
 qmlimportscanner.CONFIG = host_build
 
 !android|android_app {
-    SUBDIRS += \
-        qml \
-        qmlprofiler \
-        qmlbundle \
-        qmllint
     qtHaveModule(quick) {
-        !static: SUBDIRS += qmlscene qmlplugindump
-        qtHaveModule(widgets): SUBDIRS += qmleasing
+        SUBDIRS += qmlscene
     }
     qtHaveModule(qmltest): SUBDIRS += qmltestrunner
     contains(QT_CONFIG, private_tests): SUBDIRS += qmljs
