diff -urN qtdeclarative-opensource-src-5.2.1.old/src/qmldevtools/qmldevtools.pro qtdeclarative-opensource-src-5.2.1/src/qmldevtools/qmldevtools.pro
--- qtdeclarative-opensource-src-5.2.1.old/src/qmldevtools/qmldevtools.pro	2014-02-01 21:38:02.000000000 +0100
+++ qtdeclarative-opensource-src-5.2.1/src/qmldevtools/qmldevtools.pro	2014-02-08 16:27:35.530461798 +0100
@@ -1,4 +1,3 @@
-option(host_build)
 TARGET     = QtQmlDevTools
 QT         = core
 CONFIG += static no_module_headers internal_module
diff -urN qtdeclarative-opensource-src-5.2.1.old/tools/tools.pro qtdeclarative-opensource-src-5.2.1/tools/tools.pro
--- qtdeclarative-opensource-src-5.2.1.old/tools/tools.pro	2014-02-01 21:38:02.000000000 +0100
+++ qtdeclarative-opensource-src-5.2.1/tools/tools.pro	2014-02-08 16:15:13.200420412 +0100
@@ -1,19 +1,11 @@
 TEMPLATE = subdirs
-SUBDIRS += \
-    qmlmin \
-    qmlimportscanner
 
 qmlmin.CONFIG = host_build
 qmlimportscanner.CONFIG = host_build
 
 !android|android_app {
-    SUBDIRS += \
-        qml \
-        qmlprofiler \
-        qmlbundle
     qtHaveModule(quick) {
-        !static: SUBDIRS += qmlscene qmlplugindump
-        qtHaveModule(widgets): SUBDIRS += qmleasing
+        SUBDIRS += qmlscene
     }
     qtHaveModule(qmltest): SUBDIRS += qmltestrunner
     contains(QT_CONFIG, private_tests): SUBDIRS += qmljs
