--- old/src/plugins/platforms/eglfs/qeglfsdeviceintegration.cpp	2016-03-12 20:31:15.565690286 +0100
+++ new/src/plugins/platforms/eglfs/qeglfsdeviceintegration.cpp	2016-03-12 20:31:39.594690857 +0100
@@ -260,7 +260,11 @@
     Q_UNUSED(platformWindow);
     Q_UNUSED(size);
     Q_UNUSED(format);
-    return 0;
+    static struct mali_native_window native_window = {
+        .width = (short unsigned int)size.width(),
+        .height = (short unsigned int)size.height(),
+    };
+    return &native_window;
 }
 
 EGLNativeWindowType QEGLDeviceIntegration::createNativeOffscreenWindow(const QSurfaceFormat &format)
