--- old/src/plugins/platforms/eglfs/qeglfsintegration.cpp	2014-07-04 12:21:42.356966501 +0200
+++ old/src/plugins/platforms/eglfs/qeglfsintegration.cpp	2014-07-04 12:24:33.539372459 +0200
@@ -58,6 +58,8 @@
 #include <QtGui/QOffscreenSurface>
 #include <qpa/qplatformcursor.h>
 
+#include <qpa/qplatformtheme.h>
+
 #include <EGL/egl.h>
 
 static void initResources()
@@ -67,6 +69,17 @@
 
 QT_BEGIN_NAMESPACE
 
+
+class QPlatformThemeOlinuxino: public QPlatformTheme {
+	QVariant themeHint(ThemeHint hint) const {
+		if (hint == QPlatformTheme::PasswordMaskDelay) {
+			return QVariant(int(2000));
+		}
+		return QPlatformTheme::themeHint(hint);
+	}
+};
+
+
 QEglFSIntegration::QEglFSIntegration()
 {
     mDisableInputHandlers = qgetenv("QT_QPA_EGLFS_DISABLE_INPUT").toInt();
@@ -79,6 +92,20 @@
     QEglFSHooks::hooks()->platformDestroy();
 }
 
+QPlatformTheme *QEglFSIntegration::createPlatformTheme(const QString &name) const
+{
+	Q_UNUSED(name);
+	QPlatformTheme *theme = new QPlatformThemeOlinuxino;
+	return theme;
+}
+
+QStringList QEglFSIntegration::themeNames() const
+{
+	QStringList themeNames;
+	themeNames << "default";
+	return themeNames;
+}
+
 bool QEglFSIntegration::hasCapability(QPlatformIntegration::Capability cap) const
 {
     // We assume that devices will have more and not less capabilities
@@ -133,6 +160,8 @@
     {
     case QPlatformIntegration::ShowIsFullScreen:
         return screen()->compositingWindow() == 0;
+    case QPlatformIntegration::PasswordMaskDelay:
+        return 2000;
     default:
         return QPlatformIntegration::styleHint(hint);
     }
--- old/src/plugins/platforms/eglfs/qeglfsintegration.h	2014-07-04 12:25:48.792298113 +0200
+++ new/src/plugins/platforms/eglfs/qeglfsintegration.h	2014-07-04 12:24:47.655962650 +0200
@@ -54,6 +54,9 @@
     QEglFSIntegration();
     ~QEglFSIntegration();
 
+    QPlatformTheme *createPlatformTheme(const QString &name) const;
+    QStringList themeNames() const;
+
     void initialize() Q_DECL_OVERRIDE;
 
     bool hasCapability(QPlatformIntegration::Capability cap) const Q_DECL_OVERRIDE;
