--- old/src/platformsupport/eglconvenience/qeglplatformintegration.cpp	2016-03-12 20:54:52.338723958 +0100
+++ new/src/platformsupport/eglconvenience/qeglplatformintegration.cpp	2016-03-12 20:55:37.081725021 +0100
@@ -106,10 +106,29 @@
     if (m_display == EGL_NO_DISPLAY)
         qFatal("Could not open egl display");
 
+    QByteArray fbDev = qgetenv("QT_QPA_EGLFS_FB");
+    if (fbDev.isEmpty())
+        fbDev = QByteArrayLiteral("/dev/fb0");
+    QFile framebuffer(QString::fromLocal8Bit(fbDev));
+    framebuffer.open(QIODevice::ReadOnly);
+    QByteArray fbData;
+    if (framebuffer.isOpen()) {
+        fbData = framebuffer.readAll();
+        framebuffer.close();
+    }
+
     EGLint major, minor;
     if (!eglInitialize(m_display, &major, &minor))
         qFatal("Could not initialize egl display");
 
+    if (!fbData.isEmpty()) {
+        framebuffer.open(QIODevice::WriteOnly);
+        if (framebuffer.isOpen()) {
+            framebuffer.write(fbData);
+            framebuffer.close();
+        }
+    }
+
     m_inputContext = QPlatformInputContextFactory::create();
 
     m_vtHandler.reset(new QFbVtHandler);
