--- old/src/plugins/platforms/eglfs/qeglfshooks_stub.cpp	2013-07-03 16:49:05.264801642 +0200
+++ new/src/plugins/platforms/eglfs/qeglfshooks_stub.cpp	2013-07-03 16:49:10.114775367 +0200
@@ -251,7 +251,12 @@
     Q_UNUSED(platformWindow);
     Q_UNUSED(size);
     Q_UNUSED(format);
-    return 0;
+
+    static struct mali_native_window native_window = {
+        .width = (short unsigned int)size.width(),
+        .height = (short unsigned int)size.height(),
+    };
+    return &native_window;
 }
 
 void QEglFSHooks::destroyNativeWindow(EGLNativeWindowType window)
