--- old/src/plugins/platforms/eglfs/qeglfshooks_stub.cpp	2014-12-21 10:33:37.084268835 +0100
+++ new/src/plugins/platforms/eglfs/qeglfshooks_stub.cpp	2014-12-21 10:35:41.820259749 +0100
@@ -161,7 +161,12 @@
     Q_UNUSED(platformWindow);
     Q_UNUSED(size);
     Q_UNUSED(format);
-    return 0;
+
+    static struct mali_native_window native_window = {
+        .width = (short unsigned int)size.width(),
+        .height = (short unsigned int)size.height(),
+    };
+    return &native_window;
 }
 
 void QEglFSHooks::destroyNativeWindow(EGLNativeWindowType window)
