--- old/libconnman-qt/networkmanager.cpp	2014-10-12 23:01:31.701259368 +0200
+++ new/libconnman-qt/networkmanager.cpp	2014-10-12 23:03:01.607438971 +0200
@@ -326,45 +326,9 @@
 
 void NetworkManager::updateDefaultRoute()
 {
-    QString defaultNetDev;
-    QFile routeFile("/proc/net/route");
-    if (routeFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
-        QTextStream in(&routeFile);
-        QString line = in.readLine();
-        while (!line.isNull()) {
-            QStringList lineList = line.split('\t');
-            if (lineList.size() >= 11) {
-                if (lineList.at(1) == "00000000" && lineList.at(3) == "0003") {
-                    defaultNetDev = lineList.at(0);
-                    break;
-                }
-            }
-            line = in.readLine();
-        }
-        routeFile.close();
-    }
-    if (defaultNetDev.isNull()) {
-         QFile ipv6routeFile("/proc/net/ipv6_route");
-         if (ipv6routeFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
-             QTextStream ipv6in(&ipv6routeFile);
-             QString ipv6line = ipv6in.readLine();
-             while (!ipv6line.isNull()) {
-                 QStringList ipv6lineList = ipv6line.split(QRegExp("\\s+"));
-                 if (ipv6lineList.size() >= 10) {
-                     if (ipv6lineList.at(0) == "00000000000000000000000000000000" && ipv6lineList.at(8).endsWith("3")) {
-                         defaultNetDev = ipv6lineList.at(9).trimmed();
-                         break;
-                     }
-                     ipv6line = ipv6in.readLine();
-                 }
-             }
-             ipv6routeFile.close();
-         }
-    }
-
-    Q_FOREACH (NetworkService *service, m_servicesCache) {
-        if (service->state() == "online" || service->state() == "ready") {
-            if (defaultNetDev == service->ethernet().value("Interface")) {
+    Q_FOREACH (NetworkService *service, getServices("")) {
+        if (1) {
+            if ("wlan0" == service->ethernet().value("Interface")) {
                 if (m_defaultRoute != service) {
                     m_defaultRoute = service;
                     Q_EMIT defaultRouteChanged(m_defaultRoute);
